lib/daru/io/exporters/excelx.rb

```diff
@@ -0,0 +1,114 @@
+require 'daru/io/exporters/base'
+
+module Daru
+  module IO
+    module Exporters
+      # Excelx Exporter Class, that extends `to_excelx_string` and `write_excelx` methods to
+      # `Daru::DataFrame` instance variables
+      class Excelx < Base
+        Daru::DataFrame.register_io_module :to_excelx_string, self
+        Daru::DataFrame.register_io_module :write_excelx, self
+
+        # Initializes an Excelx Exporter instance.
+        #
+        # @param dataframe [Daru::DataFrame] A dataframe to export. Supports even dataframes
+        #   with multi-index.
+        # @param sheet [String] A sheet name, to export the dataframe into. Defaults to
+        #   'Sheet0'.
+        # @param header [Boolean] Defaults to true. When set to false or nil,
+        #   headers are not written.
+        # @param data [Boolean] Defaults to true. When set to false or nil,
+        #   data values are not written.
+        # @param index [Boolean] Defaults to true. When set to false or nil,
+        #   index values are not written
+        #
+        # @example Initializing an Excel Exporter instance
+        #   df = Daru::DataFrame.new([[1,2],[3,4]], order: [:a, :b])
+        #
+        #   #=> #<Daru::DataFrame(2x2)>
+        #   #      a   b
+        #   #  0   1   3
+        #   #  1   2   4
+        #
+        #   instance = Daru::IO::Exporters::Excelx.new(df)
+        def initialize(dataframe, sheet: 'Sheet0', header: true, data: true, index: true)
+          optional_gem 'rubyXL'
+
+          super(dataframe)
+          @data   = data
+          @index  = index
+          @sheet  = sheet
+          @header = header
```

Comment:
Hm, do we have this lot of boolean vars in other importers too? I believe that one hash `@render_parts = {data: data, headers: headers, index: index}` could be more effective and will allow to generalize a lot. In addition, looking at var names you can wrongly conclude that, say, `@index` contains dataframe's index, not "if we need to render index" boolean.